Assets/Packages/com.studio23.ss2.upmassistant/Editor/Manager/APIHandler.cs

```diff
@@ -0,0 +1,104 @@
+﻿using UnityEngine;
+using UnityEngine.Networking;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.IO;
+using Studio23.SS2.UPMAssistant.Editor.Data;
+using UnityEditor;
+
+
+namespace Studio23.SS2.UPMAssistant.Editor
+{ 
+    public static class APIHandler  
+{
+    
+    private const string APIURL = "https://api.github.com/licenses";
+    public static int SelectedLicenceIndex;
+    public static List<GitHubLicense> gitHubLicense;
+    public static bool IsDownloading = false;
+ 
+    private static string _selectedLicenceURL; 
+    
+    public static void FetchOnlineGitLicenses()
+    {
+        _selectedLicenceURL = DataHandler.LoadLicenseURLData();
+        UnityWebRequest www = UnityWebRequest.Get(APIURL);
+        www.SendWebRequest().completed += FetchLicensesCallback;
+    }
+
+    private static void FetchLicensesCallback(AsyncOperation operation)
+    {
+        UnityWebRequest www = ((UnityWebRequestAsyncOperation)operation).webRequest;
+
+        if (www.isNetworkError || www.isHttpError)
+        {
+            Debug.LogError("Error: " + www.error);
+        }
+        else
+        {
+            string jsonResponse = www.downloadHandler.text;
+            gitHubLicense = JsonConvert.DeserializeObject<List<GitHubLicense>>(jsonResponse);
+            SelectedLicenceIndex =  GetLicenseIndex(_selectedLicenceURL);
+           
+        }
+    }
+    
+   
+    public static void DownloadLicence()
+    {
+        IsDownloading = true;
+        UnityWebRequest www = UnityWebRequest.Get(DataHandler.LoadLicenseURLData());
+            www.SendWebRequest().completed += FetchLicensesCallback2;
+    }
+    private static void FetchLicensesCallback2(AsyncOperation operation)
+    {
+        UnityWebRequest www = ((UnityWebRequestAsyncOperation)operation).webRequest;
+
+        if (www.isNetworkError || www.isHttpError)
+        {
+            Debug.LogError("Error: " + www.error);
+        }
+        else
+        {
+            string jsonResponse = www.downloadHandler.text;
+            var license = JsonConvert.DeserializeObject<License>(jsonResponse);
+            var licenseFilePath =
+                Path.Combine(DataHandler.ROOT + DataHandler.LoadPackageNameData(), DataHandler.LICENSE_MD);
+            if (!File.Exists(licenseFilePath))
+            {
+                Debug.LogError("License file not found!");
+                return;
+            }
+            File.WriteAllText(licenseFilePath, string.Empty);
+            File.WriteAllText(licenseFilePath, license.body);
```

Comment:
async method should be done

Reply:
The File.WriteAllText method is a synchronous method, which means that it blocks the calling thread until the write operation is complete